### PR TITLE
Fixes #14951 - Allow matcher for host turning into managed host

### PR DESCRIPTION
**Pull request by orrabin**
I've only been able to test this with discovery, I can add a corresponding discovery pr with tests unless someone has an idea how to test this in core.

### CODE REVIEW

**Comment by lzap** [comment]
As per our IRC discussion, can you please add a test for this in discovery plugin? I will test it together.

**Comment by orrabin** [comment]
@lzap I was able to reproduce in a test in core so there is no need for a test in discovery

File: app/models/lookup_value.rb [hunk 1]
```diff
@@ -75,7 +75,8 @@ def ensure_fqdn_exists
     md = ensure_matcher(/fqdn=(.*)/)
     return md if md == true || md == false
     fqdn = md[1].split(LookupKey::KEY_DELM)[0]
-    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?)
+    return true if Host.unscoped.find_by_name(fqdn) || host_or_hostgroup.try(:new_record?) ||
+        (host_or_hostgroup.present? && host_or_hostgroup.type_changed? && host_or_hostgroup.type == "Host::Managed")
```
Comment (lzap) [inline app/models/lookup_value.rb:79]: What is the `host_or_hostgroup.present?` for? I mean, you can create discovered hosts without hostgroups just fine.

**Comment by unorthodoxgeek** [comment]
@mention-bot - can you please not tag me? 🚯

File: app/models/lookup_value.rb (hunk 1 above)
Comment (orrabin) [reply app/models/lookup_value.rb:79]: `host_or_hostgroup` is the discovered host itself, you can either create a lookup_value matcher yourself or override a parameter from the host or hostgroup in which can host_or_hostgroup will be the host/hostgroup that created an override which triggers a creation of the matcher.

**Comment by lzap** [comment]
@unorthodoxgeek https://github.com/mention-bot/how-to-unsubscribe

**Comment by lzap** [comment]
Thanks, merged.

**Pull request merged**